sound/soc/sof/ipc4-topology.c

```diff
@@ -2837,17 +2830,29 @@ static int sof_ipc4_dai_config(struct snd_sof_dev *sdev, struct snd_sof_widget *
 		gtw_attr->lp_buffer_alloc = pipeline->lp_mode;
 		fallthrough;
 	case SOF_DAI_INTEL_ALH:
+	{
+		struct sof_ipc4_alh_configuration_blob *blob;
+
+		blob = (struct sof_ipc4_alh_configuration_blob *)ipc4_copier->copier_config;
 		/*
 		 * Do not clear the node ID when this op is invoked with
 		 * SOF_DAI_CONFIG_FLAGS_HW_FREE. It is needed to free the group_ida during
 		 * unprepare.
 		 */
 		if (flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS) {
+			int i;
+
+			for (i = 0; i < SOF_IPC4_DMA_DEVICE_MAX_COUNT; i++) {
+				if (!blob->alh_cfg.mapping[i].device) {
```

Comment:
where is this reset @ranj063 ? Looks like never?

Reply:
yeah I need to think about where to reset it but in my defense, the original code never reset it either just overwrote it